tiny_dnn/core/kernels/global_avepool_grad_op.h

```diff
@@ -27,10 +28,17 @@ class GlobalAvePoolGradOp : public core::OpKernel {
     // initialize outputs
     fill_tensor(prev_delta, float_t{0});
 
-    // only internal kernel op implemented yet, so use it regardless
-    // of the specified backend engine
-    kernels::global_avepool_grad_op_internal(prev_delta, curr_delta, params,
-                                             context.parallelize());
+    const core::backend_t engine = context.engine();
+
+    if (engine == core::backend_t::avx) {
+#ifdef CNN_USE_AVX
```

Comment:
This would mean that if `engine` is set to `core::backend_t::avx` and `CNN_USE_AVX` is undefined, nothing happens? This is kinda not intuitive and also inconsistent with other layers where we fall back to internal in this case.